analysis_driver/reader/demultiplexing_parsers.py

```diff
@@ -1,140 +1,144 @@
-        if project.get('name') == 'default':
-            continue
-
-        for sample in project.findall('Sample'):
-            for barcode in sample.findall('Barcode'):
-                if project.get('name') != 'all' and barcode.get('name') == 'all':
-                    continue
-
-                for lane in barcode.findall('Lane'):
-                    all_elements.append(
-                        (
-                            project.get('name'),
-                            sample.get('name'),
-                            barcode.get('name'),
-                            lane.get('number'),
-                            lane.find('BarcodeCount').text
-                        )
-                    )
-    return all_elements
-
-
-def parse_conversion_stats(xml_file, has_barcode):
-    tree = ElementTree.parse(xml_file).getroot()
-    all_barcodes_per_lanes = []
-    all_barcodeless = []
-
-    if not has_barcode:
-        for project in tree.iter('Project'):
-            if project.get('name') == 'all':
-                continue
+def parse_json_stats(json_data, run_id):
+    """Creates an array of tuples of run elements, unknown run elements and associated metadata."""
+    all_run_elements = []
+    adapter_trimmed_by_id = {}
+
+    # Parsing conversion results, including demultiplexed results and unknown barcodes
+    for lane in json_data['ConversionResults']:
+        all_run_elements, adapter_trimmed_by_id = \
+            _parse_conversion_results_from_json_stats(run_id=run_id, lane=lane,
+                                                      all_run_elements=all_run_elements,
+                                                      adapter_trimmed_by_id=adapter_trimmed_by_id)
+
+        """ Including the values for unknown barcodes, only for multiplexed runs"""
+        if 'Undetermined' in lane:
+            all_run_elements, adapter_trimmed_by_id = \
+                _parse_undetermined_run_elements_from_json_stats(lane=lane,
+                                                                 run_id=run_id,
+                                                                 adapter_trimmed_by_id=adapter_trimmed_by_id,
+                                                                 all_run_elements=all_run_elements)
+
+    # parsing the top 10 unknown barcodes into their own array
+    unknown_run_elements = _parse_top_unknown_run_elements_from_json_stats(json_data=json_data)
+
+    return all_run_elements, unknown_run_elements, adapter_trimmed_by_id
+
+
+def _parse_conversion_results_from_json_stats(run_id, lane, all_run_elements, adapter_trimmed_by_id):
+    for sample in lane['DemuxResults']:
+        """ If the run is not multiplexed, the index_sequence will not be present in the JSON file, and the cluster
+        count needs to be retrieved from a different variable in the JSON file. The index_sequence variable will be 
+        prepopulated as None. """
+        r1, r2 = sample['ReadMetrics'][0], sample['ReadMetrics'][1]
```

Comment:
```suggestion
        r1 = sample['ReadMetrics'][0] if sample['ReadMetrics'][0]['ReadNumber'] == 1 else sample['ReadMetrics'][1]
        r2 = sample['ReadMetrics'][1] if sample['ReadMetrics'][1]['ReadNumber'] == 2 else sample['ReadMetrics'][0]
```
an no need for the `NotImplementedError` in L60

Reply:
Thanks @tcezard I'll implement this.

Reply:
@tcezard I am still leaving the error, but converting it to a value error, just in case this doesn't work -it should never not work, but I'm catering for the eventuality just the same.